server/jetstream_cluster_4_test.go

```diff
@@ -2676,3 +2677,396 @@ func TestJetStreamClusterKeyValueDesyncAfterHardKill(t *testing.T) {
+					}
+					streamLeader := getStreamDetails(t, c, accountName, str)
+					if streamLeader == nil {
+						continue
+					}
+					t.Logf("|------------------------------------------------------------------------------------------------------------------------|")
+					lstate := streamLeader.State
+					t.Logf("| %-10s | %-10s | msgs:%-10d | bytes:%-10d | deleted:%-10d | first:%-10d | last:%-10d |",
+						str, leaderSrv.String()+"*", lstate.Msgs, lstate.Bytes, lstate.NumDeleted, lstate.FirstSeq, lstate.LastSeq,
+					)
+					for _, srv := range c.servers {
+						if srv == leaderSrv {
+							continue
+						}
+						acc, err := srv.LookupAccount(accountName)
+						if err != nil {
+							continue
+						}
+						stream, err := acc.lookupStream(str)
+						if err != nil {
+							t.Logf("Error looking up stream %s on %s replica", str, srv)
+							continue
+						}
+						state := stream.state()
+
+						unsynced := lstate.Msgs != state.Msgs || lstate.Bytes != state.Bytes ||
+							lstate.NumDeleted != state.NumDeleted || lstate.FirstSeq != state.FirstSeq || lstate.LastSeq != state.LastSeq
+
+						var result string
+						if unsynced {
+							result = "UNSYNCED"
+						}
+						t.Logf("| %-10s | %-10s | msgs:%-10d | bytes:%-10d | deleted:%-10d | first:%-10d | last:%-10d | %s",
+							str, srv, state.Msgs, state.Bytes, state.NumDeleted, state.FirstSeq, state.LastSeq, result,
+						)
+					}
+				}
+				t.Logf("|------------------------------------------------------------------------------------------------------------------------| %v", nc2.ConnectedUrl())
+			}
+		}()
+	}
+
+	checkStreams := func(t *testing.T) {
+		for _, str := range streams {
+			checkFor(t, time.Minute, 500*time.Millisecond, func() error {
+				return checkState(t, c, accountName, str)
+			})
+			checkFor(t, time.Minute, 500*time.Millisecond, func() error {
+				return checkMsgsEqual(t, accountName, str)
+			})
+		}
+	}
+
+Loop:
+	for range time.NewTicker(30 * time.Second).C {
+		select {
+		case <-ctx.Done():
+			break Loop
+		default:
+		}
+		rollout := func(t *testing.T) {
+			for _, s := range c.servers {
+				// For graceful mode
+				s.lameDuckMode()
+				s.WaitForShutdown()
+				s = c.restartServer(s)
+
+				hctx, hcancel := context.WithTimeout(context.Background(), 15*time.Second)
+				defer hcancel()
+
+			Healthz:
+				for range time.NewTicker(2 * time.Second).C {
+					select {
+					case <-hctx.Done():
+					default:
+					}
+
+					status := s.healthz(nil)
+					if status.StatusCode == 200 {
+						break Healthz
+					}
+				}
+				c.waitOnClusterReady()
+				checkStreams(t)
+			}
+		}
+		rollout(t)
+		checkStreams(t)
+	}
+	wg.Wait()
+	checkStreams(t)
+}
+
+func TestJetStreamClusterKeyValueLastSeqMismatch(t *testing.T) {
+	c := createJetStreamClusterExplicit(t, "R3S", 3)
+	defer c.shutdown()
+
+	nc, js := jsClientConnect(t, c.randomServer())
+	defer nc.Close()
+
+	kv, err := js.CreateKeyValue(&nats.KeyValueConfig{
+		Bucket:   "mismatch",
+		Replicas: 3,
```

Comment:
If replicas is set to `1`, this test fails:
```
--- a/server/jetstream_cluster_4_test.go
+++ b/server/jetstream_cluster_4_test.go
@@ -3048,7 +3048,7 @@ func TestJetStreamClusterKeyValueLastSeqMismatch(t *testing.T) {

        kv, err := js.CreateKeyValue(&nats.KeyValueConfig{
                Bucket:   "mismatch",
-               Replicas: 3,
+               Replicas: 1,
        })
        require_NoError(t, err)
```
```
1 lastseq 2
    jetstream_cluster_4_test.go:3070: require error, but got none
--- FAIL: TestJetStreamClusterKeyValueLastSeqMismatch (0.52s)
```

Is that expected?

Reply:
This is a cluster specific test. So needs to be R3.

Reply:
Have created an issue and will have a fix for this shortly: https://github.com/nats-io/nats-server/issues/5840

Reply:
> This is a cluster specific test. So needs to be R3.

Makes sense, but still the observed behavior of the KV itself is different here depending on whether or not the KV happens to be clustered. All the ops are mutations too which should all go to the leader, so I'd expect R1 and R3 to behave the same in this case. Not suggesting that we change this test to R1, but that the behavior here should be consistent for both.